toolz/functoolz.py

```diff
@@ -541,3 +541,32 @@ def do(func, x):
     """
     func(x)
     return x
+
+
+@curry
+def flip(func, a, b):
+    """Call the function call with the arguments flipped.
+
+    Parameters
+    ----------
+    f : callable
+        A binary function.
+    a : any
+        The second argument to `f`.
+    b : any
+        The first argument to `f`.
+
+    Returns
```

Comment:
Do you think that the docstring should talk about the builtin function case and maybe use `isinstance` as an example. I imagine some people will say, "isn't this functools.partial?" so maybe we should talk about that?